cache/models/samples.js

```diff
@@ -56,26 +58,54 @@ const ONE = 1;
 const TWO = 2;
 const MINUS_ONE = -1;
 
+/**
+ * Parses a sample name into its separate parts, i.e. the subject absolute
+ * path and the aspect name.
+ *
+ * @param {String} name - The sample name
+ * @returns {Object} containing subject.absolutePath and aspect.name
+ * @throws {ResourceNotFoundError} if the name cannot be split in two using
+ *  sampleNameSeparator
+ */
+function parseName(name) {
+  const retval = {
+    subject: { absolutePath: undefined },
+    aspect: { name: undefined },
+  };
+  const arr = (name || '').split(sampleNameSeparator);
+  if (arr.length === 2) {
+    retval.subject.absolutePath = arr[0];
+    retval.aspect.name = arr[1];
+    return retval;
+  }
+
+  logger.error(`cache/models/samples.parseName|Invalid sample name "${name}"`);
```

Comment:
Do we also want to log the function from which the error was logged to identify the action (`/post/put / patch / delete`) that caused this to happen?

Reply:
That will get logged from the extra logging in the api controller